src/app/lib/utilities/getRemoteConfig/index.test.js

```diff
@@ -0,0 +1,85 @@
+import fetchMock from 'fetch-mock';
+
+const mockUrl =
+  'https://mock-config-endpoint?application=simorgh&service=mundo&__amp_source_origin=http://localhost';
+const mockResponse = { ads: { enabled: true } };
+fetchMock.mock(mockUrl, mockResponse);
+
+describe('getRemoteConfig', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    process.env.SIMORGH_CONFIG_URL = 'https://mock-config-endpoint';
+  });
+
+  it('should return null if enableFetchingToggles is not enabled', async () => {
+    jest.mock('#lib/config/toggles', () => ({
+      local: {
+        enableFetchingToggles: { enabled: false },
+      },
+    }));
+
+    const getRemoteConfig = await import('./index');
+    const remoteToggles = await getRemoteConfig.default('mundo');
+
+    expect(remoteToggles).toBe(null);
+  });
+
+  describe('with enableFetchingToggles enabled', () => {
+    beforeEach(() => {
+      jest.mock('#lib/config/toggles', () => ({
+        local: {
+          enableFetchingToggles: { enabled: true },
+        },
+      }));
+    });
+
+    it('should return the data received from the config enpoint', async () => {
+      const getRemoteConfig = await import('./index');
+      const remoteToggles = await getRemoteConfig.default('mundo');
+
+      expect(remoteToggles).toEqual(mockResponse);
+    });
+
+    it('should return cache if it exists', async () => {
+      const mockCache = {
+        get: jest.fn(() => true),
+      };
+
+      const getRemoteConfig = await import('./index');
+      const remoteToggles = await getRemoteConfig.default('mundo', mockCache);
+
+      expect(remoteToggles).toBe(true);
+      expect(mockCache.get).toHaveBeenCalledTimes(1);
+      expect(mockCache.get).toHaveBeenCalledWith(
+        'https://mock-config-endpoint?application=simorgh&service=mundo&__amp_source_origin=http://localhost',
+      );
+    });
+
+    it('should set cache if it does not exists', async () => {
+      const mockCache = {
+        get: jest.fn(() => undefined),
+        set: jest.fn(),
+      };
+
+      const getRemoteConfig = await import('./index');
+      const remoteToggles = await getRemoteConfig.default('mundo', mockCache);
+
+      expect(remoteToggles).toEqual(mockResponse);
+      expect(mockCache.set).toHaveBeenCalledTimes(1);
+      expect(mockCache.set).toHaveBeenCalledWith(mockUrl, mockResponse);
+    });
+
+    it('should catch response errors and return null', async () => {
```

Comment:
We could also add an assertion test for the log here.

i.e: 

```
import nodeLogger from '#testHelpers/loggerMock';

expect(nodeLogger.error).toHaveBeenCalledWith(CONFIG_FETCH_ERROR, {
  url: 'https://mock-config-endpoint?application=simorgh&service=pidgin&__amp_source_origin=http://localhost',
  error: 'Error: Unexpected response (HTTP status code 500) when requesting ${url}',
service: pidgin,
      });


```

Reply:
Updated tests to check this!